kasa/discover.py

```diff
@@ -395,6 +420,7 @@ async def discover(
         discovery_packets: int = 3,
         interface: str | None = None,
         on_unsupported: OnUnsupportedCallable | None = None,
+        on_json_discovered: OnJsonDiscoveredCallable | None = None,
```

Comment:
I think I would personally define this just next to `on_discovered` and name it `on_discovered_raw`. As the docstring below states, this will be the raw discovery response payload. This would likely also mean that we shouldn't pass our internals along as parameters to this?